app/controllers/amusement_parks_controller.rb

```diff
@@ -0,0 +1,6 @@
+class AmusementParksController < ApplicationController
+  def show
+    @amusement_park = AmusementPark.find(params[:id])
+    @mechanics = Mechanic.joins(:rides).where(rides: { amusement_park_id: @amusement_park.id }).distinct
```

Comment:
This is a good start here, but remember, anytime we're doing AR like this, this logic should live in the Model and should be tested with a model test. I would have made this method something like `def unique_mechanics` in the AmusementPark model. Then from the view, we'd call `@amusement_park.unique_mechanics`.